test/powershell/Language/Parser/Parser.Tests.ps1

```diff
@@ -154,55 +160,55 @@ Describe "ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs)"
-        foreach ($command in "1;2;3|3", '1;2;3|$(1+1)', "1;2;3|'abc'") {
-            { ExecuteCommand $command } | Should -Throw -ErrorId "ParseException"
-        }
+    $PipeIntoValueTests = @(
+        @{ Command = '1;2;3|3' }
+        @{ Command = '1;2;3|$(1+1)' }
+        @{ Command = "1;2;3|'abc'" }
+    )
+    It "Throws when you pipe into a value expression (line 238)" -TestCases $PipeIntoValueTests {
+        param($Command)
+
+        { ExecuteCommand $command } | Should -Throw -ErrorId "ParseException"
     }
 
     It "Throws an incomplete parse exception when a comma follows an expression (line 247)" {
         { ExecuteCommand "(1+ 1)," } | Should -Throw -ErrorId "IncompleteParseException"
     }
 
     It "Test that invoke has a higher precedence for a script than for an executable. (line 279)" {
-        "1">$testfile
+        Set-Content -Path $testfile -Value "1"
         $result = ExecuteCommand ". $testfile"
         $result | Should -Be 1
     }
 
     It "This test will check that a path is correctly interpreted when using '..' and '.'  (line 364)" {
-        $result = ExecuteCommand "set-location $TestDrive; get-childitem dir1\.\.\.\..\dir1\.\dir2\..\..\dir1\.\dir2"
+        $result = ExecuteCommand "Set-Location $TestDrive; Get-ChildItem dir1\.\.\.\..\dir1\.\dir2\..\..\dir1\.\dir2"
         $result.Count | Should -Be 2
         $result[0].Name | Should -BeExactly "testdirfile1.txt"
         $result[1].Name | Should -BeExactly "testdirfile2.txt"
     }
 
     It "This test will check that the parser can handle a mix of forward slashes and back slashes in the path (line 417)" {
-        $result = ExecuteCommand "get-childitem $TestDrive/dir1/./.\.\../dir1/.\dir2\../..\dir1\.\dir2"
+        $result = ExecuteCommand "Get-ChildItem $TestDrive/dir1/./.\.\../dir1/.\dir2\../..\dir1\.\dir2"
         $result.Count | Should -Be 2
         $result[0].Name | Should -BeExactly "testdirfile1.txt"
         $result[1].Name | Should -BeExactly "testdirfile2.txt"
     }
 
     It "This test checks that the asterisk globs as expected. (line 545)" {
-        $result = ExecuteCommand "get-childitem $TestDrive/dir1\dir2\*.txt"
+        $result = ExecuteCommand "Get-ChildItem $TestDrive/dir1\dir2\*.txt"
         $result.Count | Should -Be 2
         $result[0].Name | Should -BeExactly "testdirfile1.txt"
         $result[1].Name | Should -BeExactly "testdirfile2.txt"
     }
 
     It "This test checks that we can use a range for globbing: [1-2] (line 557)" {
-        $result = ExecuteCommand "get-childitem $TestDrive/dir1\dir2\testdirfile[1-2].txt"
+        $result = ExecuteCommand "Get-ChildItem $TestDrive/dir1\dir2\testdirfile[1-2].txt"
         $result.Count | Should -Be 2
         $result[0].Name | Should -BeExactly "testdirfile1.txt"
         $result[1].Name | Should -BeExactly "testdirfile2.txt"
     }
 
-    It "This test will check that escaping the $ sigil inside single quotes simply returns the $ character. (line 583)" {
```

Comment:
why was this removed?

Reply:
This was removed because the test does nothing. Look at this:
```
        $result = ExecuteCommand "'`$'"
        $result | Should -BeExactly "`$"
```
That first command causes PowerShell to _already evaluate_ the backtick escape as it's passed into the command. `ExecuteCommand` only ever sees the string `'$'`. Contrary to the expectations of whoever wrote this test, the PowerShell instance being tested will only ever see a single-quoted string containing simply the character `$`.

Now, if we actually assess the test based on what it _says_ it should be doing, we can see it's flatly wrong. Single-quoted strings simply do not process escape characters, that's sort of the point. Very simple test from any PS console:
```
'`$' -ne '$'
```

Reply:
Then we should test for that explicitly, instead of maintaining a misleading test. 🙂